corehq/apps/linked_domain/updates.py

```diff
@@ -173,16 +174,31 @@ def update_user_roles(domain_link):
         reduce=False,
     )
     local_roles_by_name = {role.name: role for role in local_roles}
+
+    # Update downstream roles based on upstream roles
     for role_def in master_results:
         role = local_roles_by_name.get(role_def['name'])
         if role:
             role_json = role.to_json()
         else:
             role_json = {'domain': domain_link.linked_domain}
 
+        role_def = copy(role_def)
+        role_def.pop('_id')
```

Comment:
possibly not really relevant here since this is a small block, but is it safe to use the same name as the the local variable assigned for each iteration that ideally should not be modifed?

Reply:
It's a small enough block that I think it's okay.

Reply:
agreed that it is a small block, as of now. Fine to punt.